ignite/metrics/classification_report.py

```diff
@@ -0,0 +1,120 @@
+import json
+from typing import Callable, Collection, Dict, List, Optional, Union
+
+import torch
+
+from ignite.metrics.fbeta import Fbeta
+from ignite.metrics.metric import Metric
+from ignite.metrics.metrics_lambda import MetricsLambda
+from ignite.metrics.precision import Precision
+from ignite.metrics.recall import Recall
+
+__all__ = ["ClassificationReport"]
+
+
+def ClassificationReport(
+    beta: int = 1,
+    output_dict: bool = False,
+    output_transform: Callable = lambda x: x,
+    device: Union[str, torch.device] = torch.device("cpu"),
+    is_multilabel: bool = False,
+    labels: Optional[List[str]] = None,
+) -> MetricsLambda:
+    r"""Build a text report showing the main classification metrics. The report resembles in functionality to
+    `scikit-learn classification_report
+    <https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html#sklearn.metrics.classification_report>`_
+    The underlying implementation doesn't use the sklearn function.
+
+    Args:
+        beta: weight of precision in harmonic mean
+        output_dict: If True, return output as dict, otherwise return a str
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+        is_multilabel: If True, the tensors are assumed to be multilabel.
+        device: optional device specification for internal storage.
+        labels: Optional list of label indices to include in the report
+
+    .. code-block:: python
+
+        rank = idist.get_rank()
+        classification_report = ClassificationReport()
+        n_iters = 80
+        s = 16
+        offset = n_iters * s
+        y_true = torch.randint(0, 2, size=(offset * idist.get_world_size(),))
+        y_preds = torch.rand(offset * idist.get_world_size(), 2)
+
+        def update(engine, i):
+            return (
+                y_preds[i * s + rank * offset: (i + 1) * s + rank * offset, :],
+                y_true[i * s + rank * offset: (i + 1) * s + rank * offset],
+            )
+
+        engine = Engine(update)
```

Comment:
@Moh-Yakoub I think this is too complex for an example of class usage. Let's propose something more simple.

Reply:
What about smth like with SSIM

```python
def process_function(engine, batch):
    # ...
    return y_pred, y

engine = Engine(process_function)
metric = ClassificationReport()
metric.attach(engine, "cr")
engine.run...
res = engine.state.metrics["cr"]
# result should be like
{
  "0": {
    "precision": 0.4891304347826087,
    "recall": 0.5056179775280899,
    "f1-score": 0.497237569060773
  },
  "1": {
    "precision": 0.5157232704402516,
    "recall": 0.4992389649923896,
    "f1-score": 0.507347254447022
  },
  "macro avg": {
    "precision": 0.5024268526114302,
    "recall": 0.5024284712602398,
    "f1-score": 0.5022924117538975
  }
}
```

Reply:
@trsvchn Great suggestion! Thanks a lot